### PR TITLE
Turbomole mos

### PR DESCRIPTION
**Pull request by oliver-s-lee**
Adds support for parsing homos, mosyms and moenergies

Please note this PR also adds the custom Turbomole attribute moirreps which is required for parsing other data (eg, excited states)

### CODE REVIEW

**Comment by shivupa** [comment]
(rebased)

File: cclib/parser/turbomoleparser.py [hunk 1]
```diff
@@ -108,8 +108,8 @@ def extract(self, inputfile, line):
         if line[3:11] == "nbf(AO)=":
             nmo = int(line.split('=')[1])
             self.set_attribute('nbasis', nmo)
-            self.set_attribute('nmo', nmo)
-
+            #self.set_attribute('nmo', nmo)
```
Comment (shivupa) [inline cclib/parser/turbomoleparser.py:111]: Should this be deleted?

**Review by shivupa — CHANGES_REQUESTED** [review]

File: cclib/parser/turbomoleparser.py (hunk 1 above)
Comment (oliver-s-lee) [reply cclib/parser/turbomoleparser.py:111]: Yes it is no longer required; there is an alternative method now for setting nmo

File: cclib/parser/turbomoleparser.py [hunk 2]
```diff
@@ -122,7 +121,71 @@ def extract(self, inputfile, line):
...
+                self.set_attribute("homos", [self.determine_homo(self.mosyms[0], orbitals)])
+            
+            if "alpha:" in line:
+                orbitals, line = self.parse_dscf_orbitals(inputfile, line)
+                if len(orbitals) > 0:
+                    homo = self.determine_homo(self.mosyms[0], orbitals)
+                    if not hasattr(self, "homos"):
+                        self.set_attribute('homos', [homo])
+                    else:
+                        self.homos[0] = homo
```
Comment (berquist) [inline cclib/parser/turbomoleparser.py:175]: Strange, do you know why this isn't covered?

**Review by berquist — APPROVED** [review]
Looks good, just one question.

File: cclib/parser/turbomoleparser.py (hunk 2 above)
Comment (oliver-s-lee) [reply cclib/parser/turbomoleparser.py:175]: The logic for this branch is only triggered if we've already parsed HOMOs before (from a previous opt step), but all our test data for unrestricted HF is single point only.

Comment (berquist) [reply cclib/parser/turbomoleparser.py:175]: That makes sense. In that case, I'm ok with this.

**Review by berquist — APPROVED** [review]

**Pull request merged**